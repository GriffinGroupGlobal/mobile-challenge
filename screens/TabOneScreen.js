import { StyleSheet, View, Text, SectionList } from "react-native";



export default function TabOneScreen() {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>EstateSpace</Text>

      
      <View style={styles.container}>
        <Text style={styles.descriptionTitle}>Mobile Dev Challenge</Text>
      </View>

      <View style={styles.descriptionContainer}>
        <SectionList
          sections={[
            {
              title: 'Requirements',
              data: [
                'Get it to work with bare minimum UI and backend integration.',
                'The code must execute.',
                'Try to limit time investment to 30-60 mins max for the initial “get it to work approach”.',
                'Share the git repo link that contains your code and readme with steps to execute.'
              ],
            }
            ,{
              title: 'Nice to Haves',
              data: [
                'Step-by-step commits : Start coding and make sure to commit logically.',
                'Tests',
                'Performance improvements',
                'Killer readme',
              ]
            }
          ]}
          renderItem={({item}) => <Text style={styles.item}>{item}</Text>}
          renderSectionHeader={({section}) => <Text style={styles.sectionHeader}>{section.title}</Text>}
          keyExtractor={(item, index) => index}
        />
      </View>

    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
  },
  descriptionContainer: {
    flex: 4,
    alignItems: "center",
    marginBottom: 20,
  },
  title: {
    fontSize: 35,
    fontWeight: "bold",
    padding: 20,
  },
  descriptionTitle: {
    fontSize: 20,
    marginTop: 20,
    borderBottomColor: 'black',
    borderBottomWidth: 2,
  },
  descriptionInfo: {
    fontSize: 20,
    marginTop: 10,
    borderBottomColor: 'black',
    borderBottomWidth: 2,
  },
  sectionHeader: {
    paddingTop: 2,
    paddingLeft: 10,
    paddingRight: 10,
    paddingBottom: 2,
    fontSize: 16,
    fontWeight: 'bold',
    backgroundColor: 'lightblue',
    alignSelf: 'center',
  },
    item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
});
