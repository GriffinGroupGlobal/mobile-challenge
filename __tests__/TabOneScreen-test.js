import React from 'react';
import TabOneScreen from './../screens/TabOneScreen';
import  renderer  from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<TabOneScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});