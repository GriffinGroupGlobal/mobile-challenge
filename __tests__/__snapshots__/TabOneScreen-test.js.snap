// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Home
  </Text>
  <View
    darkColor="rgba(255,255,255,0.1)"
    lightColor="#eee"
    style={
      Object {
        "height": 1,
        "marginVertical": 30,
        "width": "80%",
      }
    }
  />
</View>
`;
